sql/testdata/select_non_covering_index

```diff
@@ -63,3 +63,25 @@ EXPLAIN SELECT * FROM t WHERE c > 0 ORDER BY c
 0 index-join
 1 scan       t@c /1-
 1 scan       t@primary
+
+statement ok
+INSERT INTO t VALUES (7, 8, NULL)
+
+query I
+SELECT c FROM t WHERE a=7
+----
+NULL
```

Comment:
I think the change to `sql.Executor` is good defense, but there is also a bug in `scanNode.filterRow`. The loop over the visible columns should not break when a non-nullable column is found. This was the root cause of a `nil` datum being returned to the `Executor`. I'll send out a PR shortly. I think this change is still worthwhile, just move the tests to `testdata/select`.